core/src/main/java/com/digitalpebble/stormcrawler/protocol/file/FileResponse.java

```diff
@@ -0,0 +1,178 @@
+/**
+ * Licensed to DigitalPebble Ltd under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * DigitalPebble licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.digitalpebble.stormcrawler.protocol.file;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.invoke.MethodHandles;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLDecoder;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Locale;
+
+import org.apache.http.HttpStatus;
+import org.slf4j.LoggerFactory;
+
+import com.digitalpebble.stormcrawler.Metadata;
+import com.digitalpebble.stormcrawler.protocol.HttpHeaders;
+import com.digitalpebble.stormcrawler.protocol.ProtocolResponse;
+
+public class FileResponse {
+
+	/**
+	 * Adopted from Apache Nutch File Protocol implementation
+	 * 
+	 * @see https://github.com/apache/nutch/blob/master/src/plugin/protocol-file
+	 *      /src/java/org/apache/nutch/protocol/file/FileResponse.java
+	 * @see https://github.com/apache/nutch/blob/master/src/plugin/protocol-file
+	 *      /src/java/org/apache/nutch/protocol/file/File.java
+	 * 
+	 * give the url as "file://C:/folder/" to the seed.   
+	 **/
+
+	private byte[] content;
+	private int statusCode;
+	private Metadata metadata;
+	private final FileProtocol fileProtocol;
+	static final SimpleDateFormat dateFormat = new SimpleDateFormat("EEE, dd MMM yyyy HH:mm:ss zzz", Locale.US);
+	static final org.slf4j.Logger LOG = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+	public FileResponse(String u, Metadata md, FileProtocol fp)
+			throws MalformedURLException, IOException {
+		
+		fileProtocol = fp;
+		metadata = md;
+
+		URL url = new URL(u);
+
+		if (LOG.isTraceEnabled()) {

```

Comment:
checking the log level is unnecessary as parameterized messages in SLF4J  avoid the concatenation.